Source/carthageTests/VersionCommandSpec.swift

```diff
@@ -0,0 +1,62 @@
+import Quick
+import Nimble
+import CarthageKit
+import Commandant
+import Cocoa
+
+import carthage
+
+private class MockPrinter: Printer {
+	private func println() {}
+
+	private var objectToPrintln: String?
+	private func println(object: Any) {
+		if let description = (object as? CustomStringConvertible)?.description {
+			objectToPrintln = description
+		}
+	}
+
+	private var objectToPrint: String?
+	private func print(object: Any) {
+		if let description = (object as? CustomStringConvertible)?.description {
+			objectToPrint = description
+		}
+	}
+}
+
+// Because this by-default synchronous and way easier to understand than ReactiveTask
+private func runCommand(command: String, fromDirectory: String = NSProcessInfo.processInfo().environment["SourceDir"] ?? "") -> String {
+	let task = NSTask()
+	let arguments = command.componentsSeparatedByString(" ")
+	task.launchPath = arguments.first!
+	task.arguments = Array(arguments[1..<arguments.count])
+	task.currentDirectoryPath = fromDirectory
+
+	let pipe = NSPipe()
+	task.standardOutput = pipe
+	task.standardError = pipe
+	task.launch()
+	task.waitUntilExit()
+
+	return String(data: pipe.fileHandleForReading.readDataToEndOfFile(), encoding: NSUTF8StringEncoding) ?? ""
+}
+
+class VersionSpec: QuickSpec {
+    override func spec() {
+		var subject: VersionCommand!
+		var printer: MockPrinter!
+
+		beforeEach {
+			printer = MockPrinter()
+			subject = VersionCommand(printer: printer)
+		}
+
+		it("prints the output of git describe --tags on the carthage directory as the version") {
+			let versionString = runCommand("/usr/bin/git describe --tags")
+
+//			subject.run(NoOptions<CarthageError>())
```

Comment:
Interestingly enough, it actually breaks (undefined symbol errors pop up, I spent about an hour trying to track them down before giving up) if I do that, presumably because of weird interactions with ReactiveCocoa and it not properly searching for the right frameworks. Which is weird because I'm essentially using the exact same xcconfig setup as CarthageKitTests, which does that everywhere.

If I were to go into the Commandant code and just make that initializer public, those errors go away. In addition, this'll be the new behavior, why not just use that?

For reference, these are the kinds of errors I get which I use your suggestion:

```
Undefined symbols for architecture x86_64:
  "ext.Result.Result.ResultType<A where A: Result.ResultType>.value.getter : A.Value?", referenced from:
      carthageTests.VersionSpec.(spec (carthageTests.VersionSpec) -> () -> ()).(closure #2) in VersionCommandSpec.o
  "direct generic type metadata pattern for Result.Result", referenced from:
      type metadata accessor for Result.Result<Commandant.NoOptions<CarthageKit.CarthageError>, Commandant.CommandantError<CarthageKit.CarthageError>> in VersionCommandSpec.o
  "protocol witness table for <A, B where B: Swift.ErrorType> Result.Result<A, B> : Result.ResultType in Result", referenced from:
      carthageTests.VersionSpec.(spec (carthageTests.VersionSpec) -> () -> ()).(closure #2) in VersionCommandSpec.o
```